README.md

```diff
@@ -1 +1,182 @@
+
+SLURM usage is optional in this set of scripts as it is considered that many of the simple CI steps, such as compilation, will happen on a login node to optimize computing time and resource sharing. Currently, the script uses non-interactive job submission and waiting loops to ensure the correct completion of the job on the cluster.
+
+A typical use case for this series of scripts is the following:
++ build: configure, build, install all the software on the login node.
++ test: reuse the previous container to launch tests on a compute node with device access through SLURM.
++ benchmark: also reuse the previous container to launch a benchmarking job on a compute node with device access through SLURM, then delete the container.
+
+See the [usage example](#Usage-Example) for concrete details.
+
+### Dependencies
+
+There are several standard Linux commands used on top of ENROOT and SLURM commands. For some commands, the script can rely on non-standard/GNU-only options. This is for now not optimized.
+
+Always required:
++ Gitlab runner (user mode)
++ Enroot
++ Flock
++ Bash 
++ grep
+
+With SLURM:
++ sacct, squeue, scancel, sbatch, srun
++ GNU ls, wc, head, tr, cut, awk, ...
++ option extglob
+
+### Code Structure
+
+The code structure is simple, there are the standard GitLab-runner custom executor scripts:
++ `config.sh`: describes the executor;
++ `prepare.sh`: prepares the enroot container from a docker image, uses an image cache, optionally reuses existing container instead;
++ `run.sh`: either directly runs the GitLab commands on a local enroot container, or submits a job that executes everything in bulk;
++ `cleanup.sh`: delete the container if not requested otherwise, cleanup the SLURM job if needed.
+
+The main configuration variables and functions are defined in the following files:
++ `include.sh`: contains the main (non slurm) configuration options;
++ `slurm_utils.sh`: contains most functions and configurations taking for SLURM functionality.
+
+### Configuration Variables
+
+The following variables control some aspects of the script functionality. They can be set as job variables in the script or on the web pages. In the script, they need to be accessed as `${CUSTOM_ENV_<VARIABLE>}`.
+
+#### Global Options
+
+These variables are not SLURM specific and can be used in the default `ENROOT` only mode.
+
++ `CI_JOB_IMAGE` (YAML script `image:` option): a standard docker image for enroot to instantiate a container from;
++ `CI_WS`: a directory with shared data access across all nodes to be used as a workspace.
+
+Optional:
++ `USE_NAME`: instead of an automatically generated name, use a specific name for the container (and SLURM job if applicable). This name needs to be unique!
++ `NVIDIA_VISIBLE_DEVICES`: a value passed to the enroot container to control NVIDIA device visibility. When no GPU is available or used, `void` should be passed.
```

Comment:
Is it only for nvidia?

Reply:
Yes, I'm not sure yet what we can do for other drivers. By default though, it seems the enroot of bwuni (or all enroot?) always expect to see a GPU available in the container and it complains or fails when no GPU is available otherwise. A way to tell enroot to not load the GPU driver part, or not care about it, is to pass this `NVIDIA_VISIBLE_DEVICES=void`, which is really important for running on the login node or CPU only nodes.